Issue 8 individual art pages

hanminh1203 opened the pull request:
## Change Summary
Improved responsive design and refactored artwork pages to follow Next.js best practices. Centralized mock data generation into a reusable hook and added proper error handling for empty database states.

### Changes Made
- Refactored `ButtonGallery` component to use Next.js `Link` component properly
- Added placeholder SVG icon for artworks without images
- Centralized mock data generation in `useArtworkData.ts` hook
- Added error handling in `getServerSideProps` to return mock data when API fails or DB is empty
- Improved responsive layouts with proper spacing and mobile-first approach
- Fixed TypeScript type issues with Next.js route parameters and PageResult types

### Change Form

- [x] The pull request title has an issue number (#8)
- [x] The change works by "Smoke testing" or quick testing
- [ ] The change has tests - NA: Mock data hook doesn't require unit tests at this stage
- [ ] The change has documentation - NA: Changes are self-explanatory code refactoring

## Other Information
- Add module django_filters.rest_framework - need confirmation

**Testing Notes:**
- Tested with empty database - placeholder icons display correctly
- Tested navigation between gallery and individual artwork pages
- Verified responsive layout on mobile and desktop viewports

# Related issue

- Resolve #8

Karl-Sue committed:
feat: add Art model

Karl-Sue committed:
refactor: keep model register minimal

DDuu123321 committed:
Add ArtContributor model with API endpoints

hanminh1203 committed:
art-page-frontend

hanminh1203 committed:
art-page-frontend

hanminh1203 committed:
Merge branch 'issue-8/art-page' of https://github.com/codersforcauses/game-dev into issue-8/art-page

Karl-Sue committed:
refactor: keep admin register simple

Karl-Sue committed:
fix: space error

hanminh1203 committed:
Load arts from backend

hanminh1203 committed:
Change field name style

hanminh1203 committed:
Readd art field to create data

hanminh1203 committed:
Make the image height corresponding to the text

hanminh1203 committed:
Filter on backend + include contributors into art

hanminh1203 committed:
Add Pagination on backend

hanminh1203 committed:
Improve frontend to support responsive

hanminh1203 committed:
Improve resize image

hanminh1203 committed:
Back button padding

Karl-Sue committed:
feature: improve responsive layout

Karl-Sue committed:
fix: resolve merge conflicts in artwork pages

Karl-Sue committed:
feat: add Art hook

Karl-Sue committed:
feat: add Artwork hook

Karl-Sue committed:
feat: add placeholder art

hanminh1203 committed:
Fix flake8

Karl-Sue committed:
fix: match Prettier code style

hanminh1203 committed:
Fix flake8

Karl-Sue committed:
Merge pull request #37 from codersforcauses/issue-8-resolve-merge-conflict

Issue 8 resolve merge conflict

hanminh1203 committed:
Refactor code for reuseability

hanminh1203 committed:
Error message

hanminh1203 committed:
Remove mock data on Frontend

hanminh1203 committed:
Merge remote-tracking branch 'origin/main' into issue-8-Individual_art_pages

hanminh1203 committed:
Solve conflict and adapt code

hanminh1203 committed:
fix flake8 on backend

hanminh1203 committed:
Fix Prettier and type check

hanminh1203 committed:
Correct django-filter version

hanminh1203 committed:
Commit poetry.lock

hanminh1203 committed:
Commit poetry.lock

hanminh1203 committed:
Correct script order

hanminh1203 committed:
Correct script order

laurenpudz commented on client/src/pages/artwork/[id].tsx:
```diff
@@ -0,0 +1,219 @@
+      />
+    </svg>
+  </div>
+);
```
please try to import an icon from an icon library instead, it will be much more concise and should be quite easy to do. lucide-react is probably our first preference but other libraries are also ok if lucide doesn't have what we need. Feel free to check with any of the project leads if you are not sure if an icon/library would be suitable or not :)

DDuu123321 replied:
Thank you for your suggestion. I have replaced the hard-coded SVG with the 'react-social-icons' library. I have also discovered and fixed missing social media link fields in the backend ('discord_url' and 'instagram_url' are now added to the ArtContributor model).

DDuu123321 replied:
Icons for Discord and Instagram have been imported via the `lucide-react` icon library.

laurenpudz commented on server/game_dev/migrations/0005_art_artcontributor.py:
```diff
@@ -0,0 +1,68 @@
+                ),
+                ("name", models.CharField(max_length=200)),
+                ("description", models.CharField(max_length=200)),
+                ("path_to_media", models.CharField(max_length=500)),
```
This should be a django image field instead. This will allow the committee to upload photos via the django admin dash

laurenpudz reviewed (changes requested):
Just a couple of notes regarding some (mostly) frontend things I've picked up on. I'd also like to get someone else's review on the backend work you've done. This PR is looking good though, keep up the good work :)

DDuu123321 committed:
feat: implement individual art pages with full functionality

- Changed Art model from CharField to ImageField for media
- Added ImageField with upload_to='art_images/'
- Created migration 0006 for field rename
- Added discord_url and instagram_url to ArtContributor model
- Created migration 0007 for social media fields
- Updated serializers for both Art and ArtContributor models
- Updated TypeScript types for Art model
- Added artwork index and detail pages with dynamic routing
- Replaced hardcoded SVG icons with react-social-icons library
- Configured Next.js to allow localhost images
- Fixed duplicate color definitions in globals.css
- Updated Tailwind config for consistent HSL colors
- Created test data script with sample images and social links

DDuu123321 committed:
fix: remove null=True from URLField to pass flake8

- Changed discord_url and instagram_url to use blank=True with default=''
- Removed null=True to follow Django best practices for string fields
- Updated migration 0007 accordingly

DDuu123321 committed:
style: fix prettier formatting in error-modal.tsx

DDuu123321 committed:
Remove the  import and then add the  icon import: Discord and Instagram

DDuu123321 committed:
Fix import formatting in [id].tsx

SafetyInObscurity commented on server/game_dev/models.py:
```diff
@@ -22,3 +22,30 @@ class Event(models.Model):
+    instagram_url = models.URLField(max_length=500, blank=True, default='')
+
+    class Meta:
+        unique_together = ('art', 'member')
```
It's preferred to use a [UniqueConstraint](https://docs.djangoproject.com/en/6.0/ref/models/constraints/#django.db.models.UniqueConstraint), as `unique_together` may become depreciated.

SafetyInObscurity commented on server/game_dev/urls.py:
```diff
@@ -1,6 +1,13 @@
 
 urlpatterns = [
     path("events/<int:id>/", EventDetailAPIView.as_view()),
+    path('', include(router.urls)),
```
Please just add the views to the urlpatterns. We don't need to define an extra router.

SafetyInObscurity commented on server/api/settings.py:
```diff
@@ -49,6 +49,7 @@
     "django.contrib.messages",
     "django.contrib.staticfiles",
     "django_extensions",
+    "django_filters",
```
Adding new dependencies to the project should be avoided where practical and should be discussed with me. Please remove this or come have a chat about it.

SafetyInObscurity commented on client/next.config.mjs:
```diff
@@ -1,27 +1,22 @@
+        port: '8000',
+        pathname: '/media/**',
+      },
+    ],
```
Since the images are not remotely hosted we should be using `localPatterns` instead of `remotePatterns`

Karl-Sue committed:
feature: add mock data for dev

DDuu123321 committed:
chore: update migrations and frontend components

- Updated migration files formatting
- Updated artwork pages and components
- Updated go-back-button component
- Updated styles and Tailwind config
- Updated serializers

DDuu123321 committed:
style: fix prettier formatting issues in [id].tsx

DDuu123321 commented on client/src/pages/artwork/[id].tsx:
```diff
@@ -0,0 +1,219 @@
+        data-layer="Right Panel"
+        className="RightPanel flex flex-col justify-start gap-2.5 py-5"
+      >
+        <div data-layer="Frame 1163" className="Frame1163 relative">
```
Renamed all Figma-generated names to descriptive ones:
- "Frame 1163" → "Contributors Section"
- "Frame 1164" → "Contributors List"
- "Frame 1099" → "Artwork Content"
- "Frame 1162" → "Desktop Artwork Info"
- "Frame 1153" → "Description Section"

Karl-Sue committed:
feature: add link to button

Karl-Sue committed:
merge: resolve merge conflict between local and remote branch

DDuu123321 committed:
Refactor: Move social links from ArtContributor to Member model and remove django-filter dependency

DDuu123321 committed:
Fix: Remove hardcoded social media fields to comply with schema design
- Remove discord_url and instagram_url from Member model
- Remove social media fields from ArtContributor model
- Delete incorrect migration 0007 (social links on ArtContributor)
- Update ArtContributorSerializer to remove social media field references
- Change unique_together to UniqueConstraint (modern Django approach)
- Add default=True to Art.active field
- Keep ArtContributor as pure junction table (art, member, role only)
This change aligns with the original schema design where social media
links are managed through a separate SocialMedia table (in development
on another branch).
Addresses James's code review feedback.

DDuu123321 commented on server/game_dev/models.py:
```diff
@@ -22,3 +22,30 @@ class Event(models.Model):
+    member = models.ForeignKey('Member', on_delete=models.CASCADE, related_name='art_contributions')
+    role = models.CharField(max_length=100)
+    discord_url = models.URLField(max_length=500, blank=True, default='')
+    instagram_url = models.URLField(max_length=500, blank=True, default='')
```
Thank you for the code review feedback. You're absolutely right about the social media fields issue. I've reviewed the original schema design and realized my initial approach was incorrect. I've now:  Removed the hardcoded social media fields** (`discord_url` and `instagram_url`)

DDuu123321 committed:
Resolve merge conflict in Navbar: keep z-100 from main

DDuu123321 committed:
Fix: Remove django_filters import and usage from views.py

DDuu123321 committed:
Update views.py

DDuu123321 committed:
Fix: Add newline at end of views.py

DDuu123321 reviewed (commented):
Thanks for the thorough review, James. I've made the following changes:

1. Removed django-filter dependency：
Deleted the DjangoFilterBackend import and related configuration from views.py
Removed django-filter==24.3 from requirements.txt

2. Refactored database models：
Removed social media fields (discord_url, instagram_url) from the Member model
Deleted the erroneous Migration 0007 (which attempted to add social media fields to ArtContributor)
Kept ArtContributor as a pure association table containing only art, member, role
Updated related serializers to remove social media field references

3. Simplified URL configuration：
Simplified the router configuration in urls.py as requested

4. Other improvements：
Replaced unique_together with UniqueConstraint
Added default=True to Art.active

All changes have been pushed, and the code now fully conforms to the original schema design. Thank you for your careful review. During multiple rounds of modifications, I lost sight of the original design requirements and added a lot of code based on my own assumptions without consulting first. I apologize for that.

Karl-Sue committed:
refactor: remove pagination

Karl-Sue committed:
refactor: change from viewset to a standard RetrieveAPIView

Karl-Sue committed:
bug: remove leftover code:

Karl-Sue committed:
refactor: remove redundant file

Karl-Sue committed:
bug: fix the unused components error

Karl-Sue committed:
Merge branch 'main' into issue-8-Individual_art_pages

Karl-Sue committed:
Merge branch 'main' into issue-8-feature

SafetyInObscurity committed:
Merge branch 'main' into issue-8-Individual_art_pages

SafetyInObscurity committed:
Merge branch 'main' into issue-8-Individual_art_pages

laurenpudz reviewed (changes requested):
hello!! could you please resolve the conflicts between this branch and the main branch, thanks!

DDuu123321 committed:
Merge branch 'main' into issue-8-Individual_art_pages

Karl-Sue committed:
refactor: redesign artshowcase

Karl-Sue committed:
fix: styling

Karl-Sue committed:
feat: add interactive artwork cards with flip animation and responsive navigation

Karl-Sue committed:
Merge branch 'issue-8-feature' into issue-8-Individual_art_pages

Karl-Sue committed:
merge: resolve merge conflict when merging main

Karl-Sue committed:
refactor: remove discord + insta url for frontend

Karl-Sue committed:
refactor: remove basedto + page-reponse

Karl-Sue committed:
styling: using font-sans instead of font-jersey10

DDuu123321 committed:
Fix: Remove deprecated social media fields and unify font usage

DDuu123321 committed:
Merge branch 'main' into issue-8-Individual_art_pages

DDuu123321 committed:
Merge conflicting migrations

DDuu123321 committed:
Fix flake8 formatting error

Karl-Sue committed:
conflict: merge main

Karl-Sue committed:
add: migration

laurenpudz reviewed (changes requested):
Please make sure you import SVGs rather than defining inline

Karl-Sue committed:
refactor: use svg instead of inline declaration

Karl-Sue committed:
Merge remote-tracking branch 'origin/main' into issue-8-Individual_art_pages

Karl-Sue commented:
Hi @RadinMan 
Our api is actually developed separately from our frontend. If you need, you can pull branch `issue-8-merge-40`, where we make regular update on integrating front + backend. Currently, we only have 1 api `/arts/featured` that returns as followed:
```
[
    {
        "art_id": 1,
        "name": "art1",
        "description": "hello art 1",
        "media": "http://localhost:8000/media/art/download.png",
        "active": true,
        "source_game_id": 1,
        "source_game_name": "game 1",
        "contributors": [
            {
                "id": 1,
                "member_id": 1,
                "member_name": "member1",
                "role": "admin"
            }
        ],
        "showcase_description": "this is good"
    }
]
```

DDuu123321 committed:
Update jersey10 font variable in Tailwind config

DDuu123321 committed:
Merge conflicting migrations 0010

DDuu123321 committed:
Merge remote-tracking branch 'origin/main' into issue-8-Individual_art_pages

DDuu123321 committed:
Resolve poetry.lock merge conflict

DDuu123321 committed:
Update poetry.lock

Karl-Sue committed:
merge: resolve merge conflict when merging main

laurenpudz commented:
closing as addressed in PR https://github.com/codersforcauses/game-dev/pull/137
